backend/lib/plugins/auth.js

```diff
@@ -16,16 +21,46 @@ const cache = new NodeCache({
   useClones: false,
 });
 
-const checkAuth = async (req) => {
+const authSetCookieOptions = (httpOnly = true) => {
+  return {
+    domain: appDomain,
+    httpOnly,
+    path: "/",
+    maxAge: auth.cookieMaxAgeSeconds,
+    sameSite: "strict",
+    secure: env !== "dev" && httpOnly,
+  };
+};
+
+const authClearCookieOptions = () => {
+  return {
+    domain: appDomain,
+    path: "/",
+    expires: new Date(0),
+  };
+};
+
+const checkAuth = async (req, reply) => {
+  // user has logged out - clear token cookie
+  if (!(REMEMBER_COOKIE in req.cookies) && TOKEN_COOKIE in req.cookies) {
```

Comment:
so here it's only checked to delete the token cookie if necessary
wondering if it should return in that case instead of also continue below for the JWT check?

Reply:
I wanted to treat it that if a request is made without the remember cookie it's like making it without a token at all - so delete the **real** token and let jwtVerify do it's thing (i.e. throw error due to no authorization in token cookie). `optionalAuthenticate` routes, such as GET /posts, for example, will hang without other code changes if I just return, so I think the approach is good but if you have another suggestion let me know.